### DISCUSSION TITLE
Update the Text of a Pull Down Toolbar Action moves it as child of the pulldown

### DISCUSSION THREAD

**Discussion by azoitl** [eclipse-platform/eclipse.platform.ui#1273] [Q&A]
I'm currently experimenting with an existing Action form the Eclipse Platform to adjust it to my use case. For that I would like to show a text besides the image of the Toolbar action. This is in itself not hard using the  `mode="FORCE_TEXT" ` option in the plugin.xml.

However everytime I try to change the text of the Action The action is moved from the toolbar into the pull down of the toolbar entry. In the toolbar no icon and no text is shown. Only the black down arrow from the drop town toolbar entry stays. When clicking on that the pull down is showing only the entry that should be in the toolbar. The original drop-down entries sometimes show up when clicking that action.  I tried different places to update the text (i.e., in the display thread, in selectionChanged, in run) but in all cases I get the same result.

So I'm thinking that this is either a bug or not possible. 

If it is a bug what do I need to do to fix it?

If it is not possible does it mean I need to re-implement all the functionality with the newer dynamic content infrastructure?

**Comment by SarikaSinha** [comment]
@vogella @akurtakov @iloveeclipse @mickaelistria 
Anyone has any suggestions here?

**Comment by iloveeclipse** [comment]
No, just debugging.

**Reply by akurtakov** [reply]
I fully agree here.
